src/server/http_server.rs

```diff
@@ -365,3 +391,57 @@ pub fn http_server(
     });
     Ok(())
 }
+
+pub fn https_server_setup(
+    conf: &Config,
+) -> Result<(u16, TcpListener, CertifiedKey), Box<dyn Error>> {
+    // Set a process wide default crypto provider.
+    let _ = rustls::crypto::ring::default_provider().install_default();
+
+    // Generate self-signed certificate
+    let cert = generate_simple_self_signed(vec![String::from("localhost"), String::from("127.0.0.1")])?;
+
+    // Bind TCP port for HTTPS
+    let listener = TcpListener::bind(&conf.https_listen)?;
+    Ok((listener.local_addr()?.port(), listener, cert))
+}
+
+pub fn https_server(
+    pstate: Arc<AuthenticatorState>,
+    listener: TcpListener,
+    cert: CertifiedKey,
+) -> Result<(), Box<dyn Error>> {
+    // Build TLS configuration.
+    let mut server_config = ServerConfig::builder()
+        .with_no_client_auth()
+        .with_single_cert(
+            vec![cert.cert.into()],
+            PrivateKeyDer::Pkcs8(PrivatePkcs8KeyDer::from(cert.key_pair.serialize_der())),
+        )
+        .map_err(|e| e.to_string())?;
+
+    // Negotiate application layer protocols: Only HTTP/1.1 and HTTP/1.0 are allowed
+    server_config.alpn_protocols = vec![b"http/1.1".to_vec(), b"http/1.0".to_vec()];
```

Comment:
I'm not sure if `parse_get` actually supports HTTP/1.0. So far I've only ever seen 1.1, so unless you're seeing 1.0 in practise, I'd suggest making this just 1.1, because then we can at least warn the user if we encounter a situation that's untested.